NOTED

Kolum yang mempunyai lebih daripada 50% missing value:\
Alley          93.767123\
MasVnrType     59.726027\
PoolQC         99.520548\
Fence          80.753425\
MiscFeature    96.301370\
dtype: float64\


Notes data storage\
df_1\
df_cleaned = Already handle missing data / Doesnt hve duplicate id / Drop id column\
df_cleaned1 = ( aku nk buang column yg 50% missing - "( 'Alley', 'MasVnrType', 'PoolQC', 'Fence', 'MiscFeature' )\
df_selected = already select features\

new data = make prediction using this feature (dataset)

In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np

In [17]:
df_1 = pd.read_csv('train.csv')
df_1.head()

Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0   1          60       RL         65.0     8450   Pave   NaN      Reg   
1   2          20       RL         80.0     9600   Pave   NaN      Reg   
2   3          60       RL         68.0    11250   Pave   NaN      IR1   
3   4          70       RL         60.0     9550   Pave   NaN      IR1   
4   5          60       RL         84.0    14260   Pave   NaN      IR1   

  LandContour Utilities  ... PoolArea PoolQC Fence MiscFeature MiscVal MoSold  \
0         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      2   
1         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      5   
2         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      9   
3         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      2   
4         Lvl    AllPub  ...        0    NaN   NaN         NaN       0     12   

  YrSold  SaleType  SaleCondition  SalePrice  
0   2008        WD         Normal     208500  
1   2007        WD         Normal     181500  
2   2008        WD         Normal     223500  
3   2006        WD        Abnorml     140000  
4   2008        WD         Normal     250000  

[5 rows x 81 columns]

In [3]:
new_data = pd.read_csv('test.csv')
new_data.head()

Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0  1461          20       RH         80.0    11622   Pave   NaN      Reg   
1  1462          20       RL         81.0    14267   Pave   NaN      IR1   
2  1463          60       RL         74.0    13830   Pave   NaN      IR1   
3  1464          60       RL         78.0     9978   Pave   NaN      IR1   
4  1465         120       RL         43.0     5005   Pave   NaN      IR1   

  LandContour Utilities  ... ScreenPorch PoolArea PoolQC  Fence MiscFeature  \
0         Lvl    AllPub  ...         120        0    NaN  MnPrv         NaN   
1         Lvl    AllPub  ...           0        0    NaN    NaN        Gar2   
2         Lvl    AllPub  ...           0        0    NaN  MnPrv         NaN   
3         Lvl    AllPub  ...           0        0    NaN    NaN         NaN   
4         HLS    AllPub  ...         144        0    NaN    NaN         NaN   

  MiscVal MoSold  YrSold  SaleType  SaleCondition  
0       0      6    2010        WD         Normal  
1   12500      6    2010        WD         Normal  
2       0      3    2010        WD         Normal  
3       0      6    2010        WD         Normal  
4       0      1    2010        WD         Normal  

[5 rows x 80 columns]

In [4]:
np.shape(df_1)

(1460, 81)

In [9]:
missing_values = df_1.isna().sum()
missing_columns = missing_values[missing_values > 0]
missing_columns

LotFrontage      259
Alley           1369
MasVnrType       872
MasVnrArea         8
BsmtQual          37
BsmtCond          37
BsmtExposure      38
BsmtFinType1      37
BsmtFinType2      38
Electrical         1
FireplaceQu      690
GarageType        81
GarageYrBlt       81
GarageFinish      81
GarageQual        81
GarageCond        81
PoolQC          1453
Fence           1179
MiscFeature     1406
dtype: int64

In [89]:
feature_missing = [
    'LotFrontage', 'Alley', 'MasVnrType', 'MasVnrArea', 
    'BsmtQual', 'BsmtExposure', 'BsmtFinType1', 'BsmtFinType2',
    'BsmtCond', 'Electrical', 'FireplaceQu', 'GarageType', 
    'GarageYrBlt', 'GarageFinish', 'GarageQual', 'GarageCond', 
    'PoolQC', 'Fence', 'MiscFeature'
]


df_1[feature_missing].info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 19 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   LotFrontage   1201 non-null   float64
 1   Alley         91 non-null     object 
 2   MasVnrType    588 non-null    object 
 3   MasVnrArea    1452 non-null   float64
 4   BsmtQual      1423 non-null   object 
 5   BsmtExposure  1422 non-null   object 
 6   BsmtFinType1  1423 non-null   object 
 7   BsmtFinType2  1422 non-null   object 
 8   BsmtCond      1423 non-null   object 
 9   Electrical    1459 non-null   object 
 10  FireplaceQu   770 non-null    object 
 11  GarageType    1379 non-null   object 
 12  GarageYrBlt   1379 non-null   float64
 13  GarageFinish  1379 non-null   object 
 14  GarageQual    1379 non-null   object 
 15  GarageCond    1379 non-null   object 
 16  PoolQC        7 non-null      object 
 17  Fence         281 non-null    object 
 18  MiscFeature   54 non-null   

In [63]:
# Mengira peratus missing value bagi setiap kolum
missing_percentage = df_1.isnull().mean() * 100

# Mengira kolum yang mempunyai missing value melebihi 50%
missing_over_50 = missing_percentage[missing_percentage > 50]

print("Kolum yang mempunyai lebih daripada 50% missing value:")
print(missing_over_50)


Kolum yang mempunyai lebih daripada 50% missing value:
Alley          93.767123
MasVnrType     59.726027
PoolQC         99.520548
Fence          80.753425
MiscFeature    96.301370
dtype: float64


# Handle Missing value

In [24]:
df_cleaned = df_1.copy()

In [31]:
# Mengisi missing value dalam lajur numerik dengan median
df_cleaned['LotFrontage'].fillna(df_cleaned['LotFrontage'].median(), inplace=True)
df_cleaned['MasVnrArea'].fillna(df_cleaned['MasVnrArea'].median(), inplace=True)
df_cleaned['GarageYrBlt'].fillna(df_cleaned['GarageYrBlt'].median(), inplace=True)



C:\Users\User\AppData\Local\Temp\ipykernel_17012\2422570481.py:2: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df_cleaned['LotFrontage'].fillna(df_cleaned['LotFrontage'].median(), inplace=True)
C:\Users\User\AppData\Local\Temp\ipykernel_17012\2422570481.py:3: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values al

In [40]:
# Mengisi missing dalam lajur categorical dengan mode
for column in ['Alley', 'MasVnrType', 'BsmtQual', 'BsmtExposure', 'BsmtFinType1', 
                 'BsmtFinType2', 'BsmtCond', 'GarageType', 'GarageFinish', 
                 'GarageQual', 'GarageCond', 'PoolQC', 'Fence', 'MiscFeature','FireplaceQu','Electrical']:
    if df_cleaned[column].isnull().sum() / len(df_cleaned) > 0.1:
        # Jika missing value signifikan, gantikan dengan 'Unknown' ( melebihi 10%)
        df_cleaned[column].fillna('Unknown', inplace=True)
    else:
        # Jika missing value kecil, gunakan mode
        df_cleaned[column].fillna(df_cleaned[column].mode()[0], inplace=True)


C:\Users\User\AppData\Local\Temp\ipykernel_17012\328090917.py:10: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df_cleaned[column].fillna(df_cleaned[column].mode()[0], inplace=True)


In [41]:
df_cleaned[feature_missing].info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 19 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   LotFrontage   1460 non-null   float64
 1   Alley         1460 non-null   object 
 2   MasVnrType    1460 non-null   object 
 3   MasVnrArea    1460 non-null   float64
 4   BsmtQual      1460 non-null   object 
 5   BsmtExposure  1460 non-null   object 
 6   BsmtFinType1  1460 non-null   object 
 7   BsmtFinType2  1460 non-null   object 
 8   BsmtCond      1460 non-null   object 
 9   Electrical    1460 non-null   object 
 10  FireplaceQu   1460 non-null   object 
 11  GarageType    1460 non-null   object 
 12  GarageYrBlt   1460 non-null   float64
 13  GarageFinish  1460 non-null   object 
 14  GarageQual    1460 non-null   object 
 15  GarageCond    1460 non-null   object 
 16  PoolQC        1460 non-null   object 
 17  Fence         1460 non-null   object 
 18  MiscFeature   1460 non-null 

In [60]:
# Menghapuskan duplikasi berdasarkan kolum 'id'
#df_cleaned = df_cleaned.drop_duplicates(subset=['Id'])
#df_cleaned

In [59]:
# Menghapuskan kolum 'id' secara tetap dalam dataset
#df_cleaned.drop(columns=['id'], inplace=True)


In [61]:
df_cleaned

MSSubClass MSZoning  LotFrontage  LotArea Street    Alley LotShape  \
0             60       RL         65.0     8450   Pave  Unknown      Reg   
1             20       RL         80.0     9600   Pave  Unknown      Reg   
2             60       RL         68.0    11250   Pave  Unknown      IR1   
3             70       RL         60.0     9550   Pave  Unknown      IR1   
4             60       RL         84.0    14260   Pave  Unknown      IR1   
...          ...      ...          ...      ...    ...      ...      ...   
1455          60       RL         62.0     7917   Pave  Unknown      Reg   
1456          20       RL         85.0    13175   Pave  Unknown      Reg   
1457          70       RL         66.0     9042   Pave  Unknown      Reg   
1458          20       RL         68.0     9717   Pave  Unknown      Reg   
1459          20       RL         75.0     9937   Pave  Unknown      Reg   

     LandContour Utilities LotConfig  ... PoolArea   PoolQC    Fence  \
0            Lvl    AllPub    Inside  ...        0  Unknown  Unknown   
1            Lvl    AllPub       FR2  ...        0  Unknown  Unknown   
2            Lvl    AllPub    Inside  ...        0  Unknown  Unknown   
3            Lvl    AllPub    Corner  ...        0  Unknown  Unknown   
4            Lvl    AllPub       FR2  ...        0  Unknown  Unknown   
...          ...       ...       ...  ...      ...      ...      ...   
1455         Lvl    AllPub    Inside  ...        0  Unknown  Unknown   
1456         Lvl    AllPub    Inside  ...        0  Unknown    MnPrv   
1457         Lvl    AllPub    Inside  ...        0  Unknown    GdPrv   
1458         Lvl    AllPub    Inside  ...        0  Unknown  Unknown   
1459         Lvl    AllPub    Inside  ...        0  Unknown  Unknown   

     MiscFeature MiscVal MoSold  YrSold  SaleType  SaleCondition  SalePrice  
0        Unknown       0      2    2008        WD         Normal     208500  
1        Unknown       0      5    2007        WD         Normal     181500  
2        Unknown       0      9    2008        WD         Normal     223500  
3        Unknown       0      2    2006        WD        Abnorml     140000  
4        Unknown       0     12    2008        WD         Normal     250000  
...          ...     ...    ...     ...       ...            ...        ...  
1455     Unknown       0      8    2007        WD         Normal     175000  
1456     Unknown       0      2    2010        WD         Normal     210000  
1457        Shed    2500      5    2010        WD         Normal     266500  
1458     Unknown       0      4    2010        WD         Normal     142125  
1459     Unknown       0      6    2008        WD         Normal     147500  

[1460 rows x 80 columns]

# Buang column yg missing melebihi 50%

In [ ]:
df_cleaned1 = df_cleaned.copy()

In [82]:
# Buang kolum berdasarkan senarai yang mempunyai lebih daripada 50% missing value
#column_drop1 = ['Alley', 'MasVnrType', 'PoolQC', 'Fence', 'MiscFeature']
#df_cleaned1.drop(columns=column_drop1, inplace=True)

# Semak dataset selepas buang kolum
#print(df_cleaned1.head())


In [92]:
df_cleaned1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 75 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   MSSubClass     1460 non-null   int64  
 1   MSZoning       1460 non-null   object 
 2   LotFrontage    1460 non-null   float64
 3   LotArea        1460 non-null   int64  
 4   Street         1460 non-null   object 
 5   LotShape       1460 non-null   object 
 6   LandContour    1460 non-null   object 
 7   Utilities      1460 non-null   object 
 8   LotConfig      1460 non-null   object 
 9   LandSlope      1460 non-null   object 
 10  Neighborhood   1460 non-null   object 
 11  Condition1     1460 non-null   object 
 12  Condition2     1460 non-null   object 
 13  BldgType       1460 non-null   object 
 14  HouseStyle     1460 non-null   object 
 15  OverallQual    1460 non-null   int64  
 16  OverallCond    1460 non-null   int64  
 17  YearBuilt      1460 non-null   int64  
 18  YearRemo

# ENCODE for selection features

### Category

In [94]:
# Pilih hanya kolum kategori (jenis data 'object')
categorical_cols = df_cleaned1.select_dtypes(include=['object'])

# Tukarkan kolum kategori ke bentuk numerik dengan Label Encoding
encoded_categorical = categorical_cols.apply(lambda col: col.astype('category').cat.codes)

# Tambahkan SalePrice ke dalam dataset yang dienkod
encoded_categorical['SalePrice'] = df_cleaned1['SalePrice']

# Hasilkan correlation matrix dengan 'SalePrice'
correlation_with_sales_price_cat = encoded_categorical.corr()['SalePrice'].sort_values(ascending=False)

# Cetak correlation
print(correlation_with_sales_price_cat)


SalePrice        1.000000
Foundation       0.382479
CentralAir       0.251328
Electrical       0.234945
PavedDrive       0.231357
RoofStyle        0.222405
SaleCondition    0.213092
Neighborhood     0.210851
HouseStyle       0.180163
GarageCond       0.134740
RoofMatl         0.132383
ExterCond        0.117303
Functional       0.115328
Exterior2nd      0.103766
Exterior1st      0.103551
GarageQual       0.098759
Condition1       0.091155
BsmtCond         0.052563
LandSlope        0.051152
Street           0.041036
BsmtFinType2     0.033235
LandContour      0.015453
Condition2       0.007513
Utilities       -0.014314
SaleType        -0.054911
LotConfig       -0.067396
BldgType        -0.085591
BsmtFinType1    -0.092106
Heating         -0.098812
MSZoning        -0.166872
LotShape        -0.255580
BsmtExposure    -0.295407
GarageType      -0.302105
HeatingQC       -0.400178
FireplaceQu     -0.459605
GarageFinish    -0.537242
KitchenQual     -0.589189
BsmtQual        -0.618025
ExterQual   

# CARI CORR DENGN Y

### Numeric

In [93]:
# Pilih hanya kolum yang bertipe data numeric
numeric_cols = df_cleaned1.select_dtypes(include=['float64', 'int64'])

# Hasilkan correlation matrix dengan 'SalePrice'
correlation_with_sales_price = numeric_cols.corr()['SalePrice'].sort_values(ascending=False)

# Cetak correlation
print(correlation_with_sales_price)


SalePrice        1.000000
OverallQual      0.790982
GrLivArea        0.708624
GarageCars       0.640409
GarageArea       0.623431
TotalBsmtSF      0.613581
1stFlrSF         0.605852
FullBath         0.560664
TotRmsAbvGrd     0.533723
YearBuilt        0.522897
YearRemodAdd     0.507101
MasVnrArea       0.472614
Fireplaces       0.466929
GarageYrBlt      0.466754
BsmtFinSF1       0.386420
LotFrontage      0.334771
WoodDeckSF       0.324413
2ndFlrSF         0.319334
OpenPorchSF      0.315856
HalfBath         0.284108
LotArea          0.263843
BsmtFullBath     0.227122
BsmtUnfSF        0.214479
BedroomAbvGr     0.168213
ScreenPorch      0.111447
PoolArea         0.092404
MoSold           0.046432
3SsnPorch        0.044584
BsmtFinSF2      -0.011378
BsmtHalfBath    -0.016844
MiscVal         -0.021190
LowQualFinSF    -0.025606
YrSold          -0.028923
OverallCond     -0.077856
MSSubClass      -0.084284
EnclosedPorch   -0.128578
KitchenAbvGr    -0.135907
Name: SalePrice, dtype: float64


# fEATURE ANALYSIS

1. Ciri Numerik\
OverallQual (0.791): Penilaian keseluruhan kualiti rumah.\
GrLivArea (0.709): Kawasan yang boleh didiami.\
GarageCars (0.640): Kapasiti kereta di garaj.\
GarageArea (0.623): Saiz keseluruhan garaj.\
TotalBsmtSF (0.614): Saiz ruang bawah tanah.\
1stFlrSF (0.606): Kawasan tingkat pertama.\
FullBath (0.561): Jumlah bilik mandi penuh.\
YearBuilt (0.523): Tahun rumah dibina.\
MasVnrArea (0.473): Kawasan batu veneer\.
Fireplaces (0.467): Jumlah pendiangan di rumah.\


2. Ciri Kategori\
Foundation (0.382): Jenis asas rumah.\
CentralAir (0.251): Sama ada rumah ada pendingin hawa.\
RoofStyle (0.222): Gaya bumbung rumah.\
ExterQual (-0.637): Kualiti luaran rumah.\
KitchenQual (-0.589): Kualiti dapur.\
Neighborhood (0.211): Lokasi kejiranan.


In [97]:
df_cleaned1

MSSubClass MSZoning  LotFrontage  LotArea Street LotShape LandContour  \
0             60       RL         65.0     8450   Pave      Reg         Lvl   
1             20       RL         80.0     9600   Pave      Reg         Lvl   
2             60       RL         68.0    11250   Pave      IR1         Lvl   
3             70       RL         60.0     9550   Pave      IR1         Lvl   
4             60       RL         84.0    14260   Pave      IR1         Lvl   
...          ...      ...          ...      ...    ...      ...         ...   
1455          60       RL         62.0     7917   Pave      Reg         Lvl   
1456          20       RL         85.0    13175   Pave      Reg         Lvl   
1457          70       RL         66.0     9042   Pave      Reg         Lvl   
1458          20       RL         68.0     9717   Pave      Reg         Lvl   
1459          20       RL         75.0     9937   Pave      Reg         Lvl   

     Utilities LotConfig LandSlope  ... EnclosedPorch 3SsnPorch ScreenPorch  \
0       AllPub    Inside       Gtl  ...             0         0           0   
1       AllPub       FR2       Gtl  ...             0         0           0   
2       AllPub    Inside       Gtl  ...             0         0           0   
3       AllPub    Corner       Gtl  ...           272         0           0   
4       AllPub       FR2       Gtl  ...             0         0           0   
...        ...       ...       ...  ...           ...       ...         ...   
1455    AllPub    Inside       Gtl  ...             0         0           0   
1456    AllPub    Inside       Gtl  ...             0         0           0   
1457    AllPub    Inside       Gtl  ...             0         0           0   
1458    AllPub    Inside       Gtl  ...           112         0           0   
1459    AllPub    Inside       Gtl  ...             0         0           0   

     PoolArea MiscVal  MoSold  YrSold  SaleType  SaleCondition SalePrice  
0           0       0       2    2008        WD         Normal    208500  
1           0       0       5    2007        WD         Normal    181500  
2           0       0       9    2008        WD         Normal    223500  
3           0       0       2    2006        WD        Abnorml    140000  
4           0       0      12    2008        WD         Normal    250000  
...       ...     ...     ...     ...       ...            ...       ...  
1455        0       0       8    2007        WD         Normal    175000  
1456        0       0       2    2010        WD         Normal    210000  
1457        0    2500       5    2010        WD         Normal    266500  
1458        0       0       4    2010        WD         Normal    142125  
1459        0       0       6    2008        WD         Normal    147500  

[1460 rows x 75 columns]

In [102]:
df_selected = df_cleaned1.copy()

# Senarai ciri yang dicadangkan termasuk SalePrice
selected_features = [
                 # Target variable
    'OverallQual',
    'GrLivArea',
    'GarageCars',
    'GarageArea',
    'TotalBsmtSF',
    '1stFlrSF',
    'FullBath',
    'YearBuilt',
    'MasVnrArea',
    'Fireplaces',
    'Foundation',
    'CentralAir',
    'RoofStyle',
    'ExterQual',
    'KitchenQual',
    'Neighborhood',
    'SalePrice'
]

# Memilih hanya ciri yang disenaraikan dalam selected_features
df_selected = df_cleaned1[selected_features]

# Memeriksa hasil
df_selected.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 17 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   OverallQual   1460 non-null   int64  
 1   GrLivArea     1460 non-null   int64  
 2   GarageCars    1460 non-null   int64  
 3   GarageArea    1460 non-null   int64  
 4   TotalBsmtSF   1460 non-null   int64  
 5   1stFlrSF      1460 non-null   int64  
 6   FullBath      1460 non-null   int64  
 7   YearBuilt     1460 non-null   int64  
 8   MasVnrArea    1460 non-null   float64
 9   Fireplaces    1460 non-null   int64  
 10  Foundation    1460 non-null   object 
 11  CentralAir    1460 non-null   object 
 12  RoofStyle     1460 non-null   object 
 13  ExterQual     1460 non-null   object 
 14  KitchenQual   1460 non-null   object 
 15  Neighborhood  1460 non-null   object 
 16  SalePrice     1460 non-null   int64  
dtypes: float64(1), int64(10), object(6)
memory usage: 194.0+ KB


# Split X and Y

In [103]:
# Memisahkan ciri (X) dan target (y)
X = df_selected.drop('SalePrice', axis=1)
y = df_selected['SalePrice']


# Encode

In [104]:
# Periksa nilai unik dalam setiap kolum kategori
for column in ['Foundation', 'CentralAir', 'RoofStyle', 'ExterQual', 'KitchenQual', 'Neighborhood']:
    print(f"\n{column} - Unique values:")
    print(df_selected[column].unique())



Foundation - Unique values:
['PConc' 'CBlock' 'BrkTil' 'Wood' 'Slab' 'Stone']

CentralAir - Unique values:
['Y' 'N']

RoofStyle - Unique values:
['Gable' 'Hip' 'Gambrel' 'Mansard' 'Flat' 'Shed']

ExterQual - Unique values:
['Gd' 'TA' 'Ex' 'Fa']

KitchenQual - Unique values:
['Gd' 'TA' 'Ex' 'Fa']

Neighborhood - Unique values:
['CollgCr' 'Veenker' 'Crawfor' 'NoRidge' 'Mitchel' 'Somerst' 'NWAmes'
 'OldTown' 'BrkSide' 'Sawyer' 'NridgHt' 'NAmes' 'SawyerW' 'IDOTRR'
 'MeadowV' 'Edwards' 'Timber' 'Gilbert' 'StoneBr' 'ClearCr' 'NPkVill'
 'Blmngtn' 'BrDale' 'SWISU' 'Blueste']


# Aku dh sampai sini so nanti kau buat analisis mne satu Ordinal dan Norminal kemudian Baru encode.. bile dh siap.. baru siap dataset aku. dan seterusnya aku bole buat scalling then run the model. scaling hanya dilakukan kepada fitur numerik, bukan kepada data kategori. aku hanya scalling data X numeric sahaja dan Y sales price xperlu



# Scalling

# Run model and evaluate

# Make prediction using new data 